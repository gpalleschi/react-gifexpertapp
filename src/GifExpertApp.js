// GifExpertApp

import React, {useState} from 'react';
import AddCategory from './components/AddCategory';
import SetElements from './components/SetElements';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories ] = useState([]); 
    const [elements, setElements ] = useState(10); 

    return (<> 
             <h2> GifExpertApp</h2> 
	     <AddCategory setCategories={setCategories}/>
	     <SetElements setElements={setElements}/>
	     <hr/> 

	     <ol>
		     {
			     categories.map((category,i) => 
			              <GifGrid key={i} category={category} elements={elements}/>
			     )}
	     </ol>
	    </>);
}

export default GifExpertApp;