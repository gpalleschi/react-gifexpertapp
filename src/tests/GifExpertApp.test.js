import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifExpertApp from '../GifExpertApp';


describe('Prueba <GifExpertApp />', () => {


	test('Test Snapshot', () => {

		const wrapper = shallow( <GifExpertApp />);
		expect( wrapper ).toMatchSnapshot();
	})

	test('Deve mostrare una lista di categorie', () => { 

		const categories = ['One Punch','Dragon Ball'];

		const wrapper = shallow( <GifExpertApp defaultCategory={categories} />);

		// Creo lo snapshot
		expect( wrapper ).toMatchSnapshot();
		// Testo se il numero di categorie Ã¨ quello atteso
		expect( wrapper.find('GifGrid').length ).toBe(categories.length);

	})

});
