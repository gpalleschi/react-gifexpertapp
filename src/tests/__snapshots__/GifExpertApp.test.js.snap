// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifExpertApp /> Deve mostrare una lista di categorie 1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <SetElements
    setElements={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      elements={10}
      key="0"
    />
    <GifGrid
      category="Dragon Ball"
      elements={10}
      key="1"
    />
  </ol>
</Fragment>
`;

exports[`Prueba <GifExpertApp /> Test Snapshot 1`] = `
<Fragment>
  <h2>
     GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <SetElements
    setElements={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
